src/main.rs

```diff
@@ -66,10 +70,31 @@ fn write_err(err: StratisError) -> StratisResult<()> {
 }
 
 fn main() {
-
-    let engine = Rc::new(RefCell::new(SimEngine::new()));
-    // TODO: add cmdline option to specify engine
-    //  let context = Rc::new(RefCell::new(Context::new()));
+    let matches = App::new("stratis")
+        .version(&crate_version!())
+        .about("Stratis storage management")
+        .arg(Arg::with_name("debug")
+            .short("d")
+            .long("debug")
+            .help("Print additional output for debugging"))
+        .arg(Arg::with_name("sim")
+            .long("sim")
+            .help("Use simulator engine"))
+        .get_matches();
+
+    if matches.is_present("debug") {
+        unsafe { debug = true }
+    };
+
+    let engine: Box<Engine> = {
```

Comment:
Runtime engine switching seems actually like a very bad idea. I'll have to table this until I've worked with tests a bit more.